lib/appsignal/rack/generic_instrumentation.rb

```diff
@@ -27,19 +28,30 @@ def call_with_appsignal_monitoring(env)
           Appsignal::Transaction::HTTP_REQUEST,
           request
         )
+        # We need to complete the transaction if there is an exception exception inside the `call`
+        # of the app. If there isn't one and the app returns us a Rack response triplet, we let
+        # the BodyWrapper complete the transaction when #close gets called on it
+        # (guaranteed by the webserver)
+        complete_transaction_without_body = false
         begin
           Appsignal.instrument("process_action.generic") do
-            @app.call(env)
+            status, headers, obody = @app.call(env)
+            [status, headers, Appsignal::Rack::BodyWrapper.wrap(obody, transaction)]
           end
         rescue Exception => error # rubocop:disable Lint/RescueException
           transaction.set_error(error)
+          complete_transaction_without_body = true
           raise error
         ensure
-          transaction.set_action_if_nil(env["appsignal.route"] || "unknown")
+          default_action = env["appsignal.route"] || env["appsignal.action"] || "unknown"
```

Comment:
We should standardize this to only `appsignal.action` 😬 
We'll do that another time, not in this PR.

Reply:
Okay, I will disregard 😆